usr/lib/common/key.c

```diff
@@ -2153,24 +2154,14 @@ CK_RV rsa_priv_check_required_attributes(TEMPLATE *tmpl, CK_ULONG mode)
             return rc;
         }
     }
-    //
-    // PKCS #11 is flexible with respect to which attributes must be present
-    // in an RSA key.  Keys can be specified in Chinese-Remainder format or
-    // they can be specified in modular-exponent format.  Right now, I only
-    // support keys created in Chinese-Remainder format.  That is, we return
-    // CKR_TEMPLATE_INCOMPLETE if a modular-exponent key is specified.  This
-    // is allowed by PKCS #11.
-    //
-    // In the future, we should allow for creation of keys in modular-exponent
-    // format too.  This raises some issues.  It's easy enough to recognize
-    // when a key has been specified in modular-exponent format.  And it's
-    // easy enough to recognize when all attributes have been specified
-    // (which is what we require right now).  What's trickier to handle is
-    // the "middle" cases in which more than the minimum yet less than the
-    // full number of attributes have been specified.  Do we revert back to
-    // modular-exponent representation?  Do we compute the missing attributes
-    // ourselves?  Do we simply return CKR_TEMPLATE_INCOMPLETE?
-    //
+
+    /*
+     * PKCS#11 is flexible with respect to which attributes must be present
+     * in an RSA key. Keys can be specified in Chinese-Remainder format or
+     * they can be specified in modular-exponent format. Effective with version
+     * 2.40, tokens MUST also store CKA_PUBLIC_EXPONENT. This permits the
```

Comment:
Maybe a bit more clear:
... MUST store the CKA_PUBLIC_EXPONENT in any case.

Reply:
fixed